### ISSUE TITLE
HTTPS not working in payara 5.192 with Open JDK 8u241

### ISSUE THREAD

**Issue by MartinWahnschaffeSymeda** [hzi-braunschweig/SORMAS-Project#1765]
### TLDR: Workaround
Do not use a JDK above 8u202 to avoid Exceptions with `Handshaker`.

### Bug Description
This is not relevant for production servers. Those are using HTTPS with Apache and forward the request to the internal HTTP ports.

Trying to open SORMAS UI via HTTPS gives the following exception

> 
> java.lang.NoSuchMethodError: sun.security.ssl.SignatureAndHashAlgorithm.getPreferableAlgorithm(Ljava/util/Collection;Ljava/lang/String;)Lsun/security/ssl/SignatureAndHashAlgorithm;
> 	at sun.security.ssl.ServerHandshaker.trySetCipherSuite(ServerHandshaker.java:1431)
> 	at sun.security.ssl.ServerHandshaker.chooseCipherSuite(ServerHandshaker.java:1197)
> 	at sun.security.ssl.ServerHandshaker.clientHello(ServerHandshaker.java:891)
> 	at sun.security.ssl.ServerHandshaker.processMessage(ServerHandshaker.java:238)
> 	at sun.security.ssl.Handshaker.processLoop(Handshaker.java:1062)
> 	at sun.security.ssl.Handshaker$1.run(Handshaker.java:995)
> 	at sun.security.ssl.Handshaker$1.run(Handshaker.java:992)
> 	at java.security.AccessController.doPrivileged(Native Method)
> 	at sun.security.ssl.Handshaker$DelegatedTask.run(Handshaker.java:1490)
> 	at org.glassfish.grizzly.ssl.SSLUtils.executeDelegatedTask(SSLUtils.java:250)
> 	at org.glassfish.grizzly.ssl.SSLBaseFilter.doHandshakeStep(SSLBaseFilter.java:683)
> 	at org.glassfish.grizzly.ssl.SSLBaseFilter.doHandshakeStep(SSLBaseFilter.java:598)
> 	at org.glassfish.grizzly.ssl.SSLBaseFilter.handleRead(SSLBaseFilter.java:310)
> 	at org.glassfish.grizzly.filterchain.ExecutorResolver$9.execute(ExecutorResolver.java:95)
> 	at org.glassfish.grizzly.filterchain.DefaultFilterChain.executeFilter(DefaultFilterChain.java:260)
> 	at org.glassfish.grizzly.filterchain.DefaultFilterChain.executeChainPart(DefaultFilterChain.java:177)
> 	at org.glassfish.grizzly.filterchain.DefaultFilterChain.execute(DefaultFilterChain.java:109)
> 	at org.glassfish.grizzly.filterchain.DefaultFilterChain.process(DefaultFilterChain.java:88)
> 	at org.glassfish.grizzly.ProcessorExecutor.execute(ProcessorExecutor.java:53)
> 	at org.glassfish.grizzly.nio.transport.TCPNIOTransport.fireIOEvent(TCPNIOTransport.java:524)
> 	at org.glassfish.grizzly.strategies.AbstractIOStrategy.fireIOEvent(AbstractIOStrategy.java:89)
> 	at org.glassfish.grizzly.strategies.WorkerThreadIOStrategy.run0(WorkerThreadIOStrategy.java:94)
> 	at org.glassfish.grizzly.strategies.WorkerThreadIOStrategy.access$100(WorkerThreadIOStrategy.java:33)
> 	at org.glassfish.grizzly.strategies.WorkerThreadIOStrategy$WorkerThreadRunnable.run(WorkerThreadIOStrategy.java:114)
> 	at org.glassfish.grizzly.threadpool.AbstractThreadPool$Worker.doWork(AbstractThreadPool.java:569)
> 	at org.glassfish.grizzly.threadpool.AbstractThreadPool$Worker.run(AbstractThreadPool.java:549)
> 	at java.lang.Thread.run(Thread.java:748)

### Steps to Reproduce
1. Use JDK 8.241 (Zulu OpenJDK). E.g JDK 8.201 is working fine for me
2. Use payara 5.192 (looks like newer versions that still use grizzly npn 1.8.1 still face the same issue)
3. Deploy the ear and UI war to the server
4. Try to open the web app via HTTPS

### Expected Behavior
Should not crash

### System Details
* Device: PC
* SORMAS version: -
* Android version/Browser: Firefox

### Additional Details
We should check whether using the latest payara with a higher JDK version (e.g. 13) works.

**Comment by StefanKock** [comment]
This problem is relevant for PROD systems when they want to call for example the geocoding service.

**Comment by StefanKock** [comment]
Since SORMAS-Docker and some folks installing SORMAS-Project themselves ran into the problem that zulu8 latest does produce this defect, I looked deeper into this.

I fount these links explaining more about the problem:
- https://blog.payara.fish/http2-in-payara-5
- https://github.com/payara/Payara/issues/3719
- https://bugs.openjdk.java.net/browse/JDK-8213936

No newer versions of grizzly-npn-bootstrap.jar available than 1.8.1:
- Maven artifacts: https://mvnrepository.com/artifact/org.glassfish.grizzly/grizzly-npn-bootstrap
- Source Code: https://github.com/javaee/grizzly-npn and https://github.com/eclipse-ee4j/grizzly-npn

When looking on running a payara domain on ubuntu systems, I see ` -Xbootclasspath/p:/opt/payara5/glassfish/lib/grizzly-npn-bootstrap-1.8.1.jar`. 
It is not in the command line when starting my DEV system on Windows, but I proved it to be present in classpath (see below).

When writing grizzly-npn-bootstrap-1.8.1 without conditions into domain.xml (`<jvm-options>-Xbootclasspath/p:${com.sun.aas.installRoot}/lib/grizzly-npn-bootstrap-1.8.1.jar</jvm-options>`), it works with `zulu8.33.0.1-jdk8.0.192` but not with `zulu8.44.0.9-jdk8.0.242`. When leaving it out, it causes `java.lang.NoClassDefFoundError: org/glassfish/grizzly/npn/AlpnServerNegotiator at org.glassfish.grizzly.config.GenericGrizzlyListener.configureHttp2Support(GenericGrizzlyListener.java:806)` (so it is needed).

Conclusion: The version switch for java 8uXXX version seems to work properly, but got incompatible again. Unfortinulately there is no newer version of `grizzly-npn-bootstrap`. Payara 5.201 does not seem to a fix for this (at least not in the `domain.xml`).

I see possible fixes right now:
1. Freeze zulu8 somewhere below JDK 8u242.
2. Produce a fixed grizzly-npn-bootstrap ourselves.
3. Move forward to JDK 11 and Payara 5.194+ (see https://blog.payara.fish/jdk-11-support-available-in-payara-platform-194) to avoid problems with HTTPS/2 and JDK 8.

**Comment by HolgerReiseVSys** [comment]
The underlying issue seems to be fixed in Payara 5.2020.2:
https://github.com/payara/Payara/pull/4599
https://docs.payara.fish/community/docs/5.2020.2/release-notes/release-notes-2020-2.html#_bug_fixes

A quick test with the current Zulu 8.46.0.19 (8u252b14) was successful.
As Zulu 11 seems to be working, too, we should consider switching the servers to the latest LTS.

**Comment by StefanKock** [comment]
Documented by @MateStrysewskeSym in SERVER_SETUP.md, which versions are to be used. Solving the problem that we are stuck in versions is supposed to be solved by #2453.
